contacts/views.py

```diff
@@ -0,0 +1,164 @@
+from django.contrib.auth.decorators import login_required
+from django.shortcuts import render
+from django.http import HttpResponse, HttpResponseNotFound
+from django.http import HttpResponseRedirect
+from contacts.models import Contact, Email, Number
+from django.contrib.auth.models import User
+
+
+def index(request):
+    return render(request, 'contacts/index.html')
+
+
+def insert(request):
+    return render(request, 'contacts/test.html')
+
+
+def add_contact(request):
```

Comment:
Add authentication on this route. In this way any post request could be catered by this code, which is a security loophole.

Reply:
i have done this using decorator of login_required
`@login_required(login_url='/')`